src/api.rs

```diff
@@ -5,54 +5,58 @@ use rocket::http::Status;
 use rocket::response::*;
 use rocket::State;
 use rocket_contrib::JSON;
-use std::sync::*;
+use std::time::*;
 
 /// The response sent back from the `/register-player` endpoint.
 #[derive(Debug, Serialize)]
 pub struct RegisterPlayerResponse {
     /// The `PlayerId` that was generated for the new player.
     pub id: PlayerId,
     pub username: String,
+    pub balls: usize,
 }
 
 /// Generates a `PlayerId` for a new player.
 // TODO: Allow players to specify a username when registering.
 #[get("/register-player")]
 pub fn register_player(
     player_id_generator: State<PlayerIdGenerator>,
-    scoreboard: State<Mutex<Scoreboard>>,
-    usernames: State<Mutex<Usernames>>,
+    players: State<PlayerMap>,
     broadcaster: State<HostBroadcaster>,
 ) -> JSON<RegisterPlayerResponse>
 {
-    let player_id = player_id_generator.next_id();
+    let id = player_id_generator.next_id();
     let username = game::generate_username();
+    let balls = 10;
 
-    // Add the username to the Usernames map
-    {
-        let mut usernames = usernames.lock().expect("Usernames mutex was poisoned");
-        let old = usernames.insert(player_id, username.clone());
-        assert_eq!(None, old, "Player ID was registered twice");
+    let player = Player {
```

Comment:
Oh yeah, I forgot to mention that I removed all the separate maps for scoreboard and usernames and such. They were making the code harder to work with and were a speculative optimization that we probably don't need yet in practice. So I've unified everything into `Player` for now, and we can split things back out if accessing the player map becomes a bottleneck.